packages/eyeglass/src/util/strings.ts

```diff
@@ -1,28 +1,95 @@
-"use strict";
+import { SassImplementation, isSassString, SassString, SassValue, toString } from "./SassImplementation";
+import { inspect } from "util";
 
-var rUnquote = /^("|')(.*)\1$/;
-var rPlaceholders = /\${([^}]+)}/g;
+let rUnquote = /^("|')(.*)\1$/;
+let rPlaceholders = /\${([^}]+)}/g;
 
-function isSassValue(item) {
-  return !!(item && item.getValue && item.constructor);
+export function unquote(sass: SassImplementation, string: string | SassValue): SassString {
+  if (typeof string === "string") {
+    return sass.types.String(string.replace(rUnquote, "$2"));
+  }
+  if (!isSassString(sass, string)) {
+    throw new Error(`Expected: Sass String. Got: ${inspect(string)}`)
+  }
+  if (rUnquote.test(string.getValue())) {
+    return sass.types.String(RegExp.$2);
+  } else {
+    return string;
+  }
+}
+
+export function unquoteJS(sass: SassImplementation, string: string | SassValue): string {
+  if (typeof string === "string") {
+    return string.replace(rUnquote, "$2");
+  }
+  if (!isSassString(sass, string)) {
+    throw new Error(`Expected: Sass String. Got: ${inspect(string)}`)
+  }
+  if (rUnquote.test(string.getValue())) {
+    return RegExp.$2;
+  } else {
+    return string.getValue();
+  }
 }
 
-export function unquote(string) {
-  if (isSassValue(string)) {
-    return string.constructor(unquote(string.getValue()));
+export function quoteSass(sass: SassImplementation, string: string | SassValue): SassString;
+export function quoteSass(sass: SassImplementation, string: undefined): undefined;
```

Comment:
I think this signature can just be

```ts
export function quoteSass(sass: SassImplementation): undefined;
```

Reply:
@mike-north I was pretty careful about this during the coding, actually. I think my code already represents my intent on this.